packages/api/src/folder/service.ts

```diff
@@ -0,0 +1,69 @@
+import createError from "http-errors";
+import { logger } from "@stela/logger";
+import { db } from "../database";
+import type {
+  FolderColumnsForUpdate,
+  FolderRow,
+  PatchFolderRequest,
+} from "./models";
+import { requestFieldsToDatabaseFields } from "./helper";
+
+export const getFolderById = async (requestQuery: {
+  folderId: string;
+}): Promise<FolderRow> => {
+  const folder = await db
+    .sql<FolderRow>("folder.queries.get_folder_by_id", {
+      folderId: requestQuery.folderId,
+    })
+    .catch((err) => {
+      logger.error(err);
+      throw new createError.InternalServerError(
+        "failed to retrieve featured archives"
+      );
+    });
+
+  if (folder.rows[0] === undefined) {
+    throw new createError.NotFound("Folder not found.");
+  }
+
+  return folder.rows[0];
+};
+
+function buildPatchQuery(patchFolderRequest: FolderColumnsForUpdate): string {
+  const updates = Object.entries(patchFolderRequest)
+    .filter(([key, value]) => value !== undefined && key !== "folderId")
+    .map(([key, _]) => `${key} = :${key}`);
+
+  if (updates.length === 0) {
+    throw new createError.BadRequest("Request cannot be empty");
+  }
+
+  const query = `
+    UPDATE folder
+    SET ${updates.join(", ")}
+    WHERE folderid = :folderId
+    RETURNING folder.folderid AS "folderId"
+  `;
+
+  return query.trim();
+}
+
+export const patchFolder = async (
+  folderId: string,
+  folderData: PatchFolderRequest
+): Promise<string> => {
+  const columnsForUpdate = requestFieldsToDatabaseFields(folderData, folderId);
+  const query = buildPatchQuery(columnsForUpdate);
+
+  const result = await db
+    .query<FolderRow>(query, columnsForUpdate)
```

Comment:
What we're getting back here isn't actually a FolderRow, is it?

Reply:
It is a folder row object but with just the folderId property. Do you think having a new interface with with just this property is needed?

Reply:
`FolderRow` has some required properties that this won't, right? I think because we're returning such a small object, and we're not using that object outside this function, I'd just do
`.query<{folderId: string}>(query, columnsForUpdate)`